client/my-sites/activity/filterbar/text-search-control.tsx

```diff
@@ -0,0 +1,51 @@
+import { useMobileBreakpoint } from '@automattic/viewport-react';
+import { TextControl } from '@wordpress/components';
+import { useState } from '@wordpress/element';
+import { useTranslate } from 'i18n-calypso';
+import React, { FunctionComponent } from 'react';
+import { useDispatch } from 'react-redux';
+import { updateFilter } from 'calypso/state/activity-log/actions';
+import { CalypsoDispatch } from 'calypso/state/types';
+
+interface Props {
+	siteId: number;
+
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	filter: any;
+}
+
+const TextSearchControl: FunctionComponent< Props > = ( { siteId, filter } ) => {
+	const translate = useTranslate();
+	const isMobile = useMobileBreakpoint();
+
+	const [ text, setText ] = useState( filter.textSearch || '' );
+	const dispatch = useDispatch() as CalypsoDispatch;
+	const onKeyDown = ( event: React.KeyboardEvent< HTMLInputElement > ) => {
+		const { value } = event.currentTarget;
+
+		if ( event.key === 'Enter' ) {
+			dispatch( updateFilter( siteId, { textSearch: value } ) );
+		}
+	};
+
+	const onChange = ( value: string ) => {
+		setText( value );
+	};
+
+	const placeholder = isMobile
+		? translate( 'Search posts' )
+		: translate( 'Search posts by ID, title or author' );
+
+	return (
+		<TextControl
```

Comment:
For a second iteration, we are going considering to use the [Search component](https://github.com/Automattic/wp-calypso/blob/f16259d92fc7731b534d06686be270c4fda9a36c/client/components/search/README.md).